ui/apps/platform/src/Containers/Navigation/NavigationSideBar/NavigationSideBar.tsx

```diff
@@ -36,10 +35,14 @@ const platformConfigurationPaths = [
     systemHealthPath,
 ];
 
-function NavigationSideBar(): ReactElement {
+type Props = {
+    hasReadAccess: (resourceName: ResourceName) => boolean;
+    isFeatureFlagEnabled: (envVar: string) => boolean;
+};
```

Comment:
A thought came to my mind about defining types like this. If we made any changes to the `hasReadAccess` or `isFeatureFlagEnabled` function, we would have to make those changes here as well, right? Would it be better to, for example with `hasReadAccess`, get that function's type from the `usePermission` hook and use it here? If any changes happen to that function and type, we can make the changes to the type in that same `usePermissions` hook and not worry about going to each file that imports it to modify the types separately